returning a closure used in runtime

### The following program `sample.go` triggers an unexpected result

```go
package main

import (
    "fmt"
    "io"
    "net/http"
    "net/http/httptest"
)   

func mock(name string) http.HandlerFunc {
    return func(rw http.ResponseWriter, req *http.Request) {
        fmt.Fprint(rw, "Hello ", name)
    }   
}   

func client(uri string) {
    resp, err := http.Get(uri)
    if err != nil {
        panic(err)
    }   
    body, err := io.ReadAll(resp.Body)
    if err != nil {
        panic(err)
    }   
    fmt.Println(string(body))
}   

func main() {
    mux := http.NewServeMux()
    server := httptest.NewServer(mux)
    defer server.Close()
    mux.Handle("/", mock("foo"))
    client(server.URL)
}
```


### Expected result

```console
$ go run ./sample.go
Hello foo
```


### Got

```console
$ yaegi ./sample.go
./sample.go:11:9: panic
./sample.go:29:9: panic
run: reflect.Set: value of type *interp.node is not assignable to type http.HandlerFunc
goroutine 1 [running]:
runtime/debug.Stack()
	/Users/marc/sdk/go1.17.5/src/runtime/debug/stack.go:24 +0x88
github.com/traefik/yaegi/interp.(*Interpreter).Execute.func1(0x14000249ae0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/program.go:131 +0x8c
panic({0x105282d00, 0x1400056c1f0})
	/Users/marc/sdk/go1.17.5/src/runtime/panic.go:1038 +0x21c
github.com/traefik/yaegi/interp.runCfg.func1(0x140001dab00, 0x14000554360, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:194 +0x164
panic({0x105282d00, 0x1400056c1f0})
	/Users/marc/sdk/go1.17.5/src/runtime/panic.go:1038 +0x21c
github.com/traefik/yaegi/interp.runCfg.func1(0x140001dabb0, 0x1400054e000, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:194 +0x164
panic({0x105282d00, 0x1400056c1f0})
	/Users/marc/sdk/go1.17.5/src/runtime/panic.go:1038 +0x21c
reflect.Value.assignTo({0x105365040, 0x140001b0d60, 0x196}, {0x10509e7b9, 0xb}, 0x1052d7460, 0x0)
	/Users/marc/sdk/go1.17.5/src/reflect/value.go:2797 +0x2cc
reflect.Value.Set({0x1052d7460, 0x140001b0d30, 0x193}, {0x105365040, 0x140001b0d60, 0x196})
	/Users/marc/sdk/go1.17.5/src/reflect/value.go:1905 +0xa4
github.com/traefik/yaegi/interp._return.func1(0x140001dabb0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:2375 +0x224
github.com/traefik/yaegi/interp.runCfg(0x1400054e000, 0x140001dabb0, 0x140004aaea0, 0x14000555b00)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:202 +0x268
github.com/traefik/yaegi/interp.call.func9(0x140001dab00)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:1389 +0x7c0
github.com/traefik/yaegi/interp.runCfg(0x14000554360, 0x140001dab00, 0x140005538c0, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:202 +0x268
github.com/traefik/yaegi/interp.(*Interpreter).run(0x14000205b00, 0x140005538c0, 0x140001da8f0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:121 +0x3d4
github.com/traefik/yaegi/interp.(*Interpreter).Execute(0x14000205b00, 0x1400056ac30)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/program.go:157 +0x1e0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0x14000205b00, {0x1400054c6c0, 0x223}, {0x16b5f77a8, 0x15}, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/interp.go:615 +0x98
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0x14000205b00, {0x16b5f77a8, 0x15})
	/Users/marc/go/src/github.com/traefik/yaegi/interp/interp.go:508 +0xe8
main.runFile(0x14000205b00, {0x16b5f77a8, 0x15}, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:153 +0xb0
main.run({0x140001b6010, 0x1, 0x1})
	/Users/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:116 +0xe14
main.main()
	/Users/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/yaegi.go:144 +0x384
```


### Yaegi Version

v0.11.1

### Additional Notes

Issue found when running yaegi on https://github.com/dariopb/traefikServiceFabricPlugin

--- a/_test/issue-1333.go
+++ b/_test/issue-1333.go
@@ -1,0 +1,37 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+)
+
+func mock(name string) http.HandlerFunc {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		fmt.Fprint(rw, "Hello ", name)
+	}
+}
+
+func client(uri string) {
+	resp, err := http.Get(uri)
+	if err != nil {
+		panic(err)
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(string(body))
+}
+
+func main() {
+	mux := http.NewServeMux()
+	server := httptest.NewServer(mux)
+	defer server.Close()
+	mux.Handle("/", mock("foo"))
+	client(server.URL)
+}
+
+// Output:
+// Hello foo

--- a/interp/run.go
+++ b/interp/run.go
@@ -2375,9 +2375,13 @@ func _return(n *node) {
 			}
 			values[i] = genValueInterface(c)
 		case valueT:
-			if t.rtype.Kind() == reflect.Interface {
+			switch t.rtype.Kind() {
+			case reflect.Interface:
 				values[i] = genInterfaceWrapper(c, t.rtype)
-				break
+				continue
+			case reflect.Func:
+				values[i] = genFunctionWrapper(c)
+				continue
 			}
 			fallthrough
 		default:

